packages/flutter/lib/src/foundation/constants.dart

```diff
@@ -78,3 +78,15 @@ const double precisionErrorTolerance = 1e-10;
 /// * [dart:io.Platform], a way to find out the browser's platform that is not
 ///   overridable in tests.
 const bool kIsWeb = bool.fromEnvironment('dart.library.js_util');
+
+/// A constant that is true if the application was compiled to WebAssembly.
+///
+/// See also:
+///
+/// * [defaultTargetPlatform], which is used by themes to find out which
+///   platform the application is running on (or, in the case of a web app,
+///   which platform the application's browser is running in). Can be overridden
+///   in tests with [debugDefaultTargetPlatformOverride].
+/// * [dart:io.Platform], a way to find out the browser's platform that is not
+///   overridable in tests.
+const bool kIsWasm = kIsWeb && bool.fromEnvironment('dart.library.ffi');
```

Comment:
https://github.com/dart-lang/sdk/commit/6b71aa1f3e997fcbbb19930a1c029ffcebf33d67 adds a proper environment variable to detect WASM - is there a difference?

Reply:
The dart version will only work on Dart SDK version `3.5.0-185.0.dev` and above.

I'm guessing that would be a better way to do this, though.

Reply:
Yep, tested on .3 (included with latest Flutter beta) (I'm pretty sure at least), but didn't seem to work properly. Maybe I was doing something wrong though? Or there's an issue with its implementation in Dart?

Reply:
dart `dart --version` for me

Reply:
Can't right now, but I'm 99% sure it was that version. I've been tracking that commit (I didn't realise this constant existed) and waiting for it to arrive, and I know for a fact I did a flutter upgrade. So it should be .3 and that should be what's on the Flutter beta atm.
I can try again tomorrow.